include/nbl/video/utilities/IGPUObjectFromAssetConverter.h

```diff
@@ -1525,20 +1549,55 @@ inline created_gpu_object_array<asset::ICPUImageView> IGPUObjectFromAssetConvert
     core::vector<size_t> redirs = eliminateDuplicatesAndGenRedirs(cpuDeps);
 
     auto gpuDeps = getGPUObjectsFromAssets<asset::ICPUImage>(cpuDeps.data(), cpuDeps.data() + cpuDeps.size(), _params);
-
+    const auto physDev = _params.device->getPhysicalDevice();
+    const auto& optimalUsages = physDev->getImageFormatUsagesOptimalTiling();
+    const auto& linearUsages = physDev->getImageFormatUsagesLinearTiling();
     for (ptrdiff_t i = 0; i < assetCount; ++i)
     {
         if (gpuDeps->begin()[redirs[i]])
         {
-            const asset::ICPUImageView::SCreationParams& cpuparams = _begin[i]->getCreationParameters();
+            const auto& cpuParams = _begin[i]->getCreationParameters();
+
             IGPUImageView::SCreationParams params = {};
-            params.flags = static_cast<IGPUImageView::E_CREATE_FLAGS>(cpuparams.flags);
+            params.flags = static_cast<IGPUImageView::E_CREATE_FLAGS>(cpuParams.flags);
+            params.viewType = static_cast<IGPUImageView::E_TYPE>(cpuParams.viewType);
             params.image = (*gpuDeps)[redirs[i]];
-            params.viewType = static_cast<IGPUImageView::E_TYPE>(cpuparams.viewType);
-            params.format = params.image->getCreationParameters().format;
-            memcpy(&params.components, &cpuparams.components, sizeof(params.components));
-            params.subresourceRange = cpuparams.subresourceRange;
-            params.subresourceRange.levelCount = (*gpuDeps)[redirs[i]]->getCreationParameters().mipLevels - params.subresourceRange.baseMipLevel;
+            const auto& gpuImgParams = params.image->getCreationParameters();
+            // override the view's format if the source image got promoted, a bit crude, but don't want to scratch my head about how to promote the views and guess semantics
+            const bool formatGotPromoted = asset::getFormatClass(cpuParams.format)!=asset::getFormatClass(gpuImgParams.format);
+            params.format = formatGotPromoted ? gpuImgParams.format:cpuParams.format;
```

Comment:
@Erfan-Ahmadi corrected the thing that would wipe away all views' formats when using Asset Converter